pages/instructorAssessmentQuestionManualGrading/instructorAssessmentQuestionManualGrading.ejs

```diff
@@ -2,18 +2,272 @@
+              $(e.target).find('input[name=batch_action_data]')
+                  .val(JSON.stringify($(submitter).data('batch-action') || {}));
+          }
+          $.post($(e.target).attr('action'), $(e.target).serialize(), function(data) {
+              $('#grading-table').bootstrapTable('refresh');
+          }, 'json').fail(function(data) {
+              // TODO Better user notification of update failure
+          });
+      }
+
+      function gradingTagDropdown() {
+          const course_staff = JSON.parse('<%- JSON.stringify(course_staff) %>') || [];
+          const dropdown = $('<div class="dropdown ml-1">');
+          const button = $('<button class="btn btn-secondary dropdown-toggle grading-tag-button" data-toggle="dropdown" name="status" disabled>').appendTo(dropdown);
+          button.text(' Tag for grading').prepend($('<i class="fas fa-tags">'));
+
+          const menu = $('<div class="dropdown-menu dropdown-menu-right">').appendTo(dropdown);
+          menu.append('<div class="dropdown-header">Tag as requiring grading</div>');
+
+          course_staff.forEach((grader) => {
+              const item_grader = $('<button class="dropdown-item" type="submit">').appendTo(menu);
+              item_grader.attr('data-batch-action', JSON.stringify({
+                  requires_manual_grading: true,
+                  assigned_grader: grader.user_id,
+              }));
+              item_grader.text(` Assign to: ${grader.name || ''} (${grader.uid})`);
+              item_grader.prepend('<i class="fas fa-user-tag"></i>');
+          });
+
+          const item_no_grader = $('<button class="dropdown-item" type="submit">').appendTo(menu);
+          item_no_grader.attr('data-batch-action', JSON.stringify({
+              requires_manual_grading: true,
+              assigned_grader: null,
+          }));
+          item_no_grader.text(` No grader assigned`).prepend('<i class="fas fa-tag"></i>');
+
+          menu.append('<div class="dropdown-divider">');
+
+          const item_graded = $('<button class="dropdown-item" type="submit">').appendTo(menu);
+          item_graded.attr('data-batch-action', JSON.stringify({
+              requires_manual_grading: false,
+          }));
+          item_graded.text(` Tag as graded`).prepend('<i class="fas fa-check-square"></i>');
+
+          return $('<div>').append(dropdown).html();
+      }
+
+      function updateGradingTagButton() {
+          $('.grading-tag-button').prop('disabled', !$('#grading-table').bootstrapTable('getSelections').length);
+      }
+
+      //<% include('../partials/pointsFormatter'); %>
+      window.getStringFromFloat = <%- getStringFromFloat.toString() %>;
+
+      function pointsFormatter(points, row) {
+          let text = `${getStringFromFloat(row.points)} <small>/<span class="text-muted"><%= max_points %></span></small> `, button = '';
+          //<% if (authz_data.has_course_instance_permission_edit) { %>
```

Comment:
the comment there is mostly to avoid having the text editor trying to interpret it as weird javascript. With the comment, emacs just treats it as a comment and doesn't try to mess up the rest of formatting.

Reply:
Maybe something like this?

https://github.com/PrairieLearn/PrairieLearn/blob/73803d56ebbf5f8f8816863d836fd3653aa0bca7/pages/instructorAssessmentQuestionManualGrading/instructorAssessmentQuestionManualGrading.ejs#L171

Reply:
That's definitely better! In the long term, I still want zero EJS interpolation in JavaScript, but that can certainly wait.

Reply:
Moved it to a script-level variable for client-side check.